staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -365,9 +365,13 @@ func (az *Cloud) serviceOwnsFrontendIP(fip network.FrontendIPConfiguration, serv
 			klog.Warningf("serviceOwnsFrontendIP: unexpected error when finding match public IP of the service %s with loadBalancerLP %s: %v", service.Name, loadBalancerIP, err)
 			return false, isPrimaryService, nil
 		}
-
-		if pip != nil && pip.ID != nil && pip.PublicIPAddressPropertiesFormat != nil && pip.IPAddress != nil {
-			if strings.EqualFold(*pip.ID, *fip.PublicIPAddress.ID) {
+		if pip != nil &&
+			pip.ID != nil &&
+			pip.PublicIPAddressPropertiesFormat != nil &&
+			pip.IPAddress != nil &&
+			fip.FrontendIPConfigurationPropertiesFormat != nil &&
+			fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress != nil {
```

Comment:
Did you want to also verify fip.PublicIPAddress != nil ?

Reply:
yes, but `fip.FrontendIPConfigurationPropertiesFormat.PublicIPAddress` is more clear for reading

Reply:
The funny thing is that I also was confused why fip.PublicIPAddress was not checked - is that not the real problem in the code on the next line?  Either that code should also used the "more clear" syntax or some comment should be here about how both end up being the same pointer since that is very much not clear from reading this code in this file.